continuous_integration/environment-3.9.yml

```diff
@@ -0,0 +1,19 @@
+name: dask-image-testenv
+
+channels:
+  - conda-forge
+
+dependencies:
+  - python=3.9.*
+  - pip>=19.3.1
+  - wheel>=0.33.6
+  - coverage>=4.5.4
+  - flake8>=3.7.9
+  - pytest>=5.3.1
+  - pytest-flake8>=1.0.4
+  - dask>=2.8.1
+  - numpy>=1.17.3
+  - scipy>=1.3.2
+  - scikit-image>=0.16.2
+  - pims>=0.4.1
+  - slicerator
```

Comment:
I've unpinned dependencies for 3.9, as the explicit pinning was causing issues with the environment solve; is there any reason to keep the pinnings on the 3.7 / 3.8 environments?

Reply:
Just to double check, you've removed the pinned version from slicerator here, and that's the only change? I'm ok with that.

I would prefer to keep the pinnings on the older 3.8/3.8 environments, on the basis of "if it ain't broke, don't fix it". That's kind of a conservative stance, but we haven't received any reports of this causing issues for people so I'm not inclined to fiddle with that.

Reply:
We should give @jakirkham a day or two to voice an objection if he has one (I don't *think* that's especially likely, but I certainly don't want to speak for him here, and don't have all the background context he does)

Reply:
Thanks for clarifying! In that case, I think we *do* want to pin the dependencies for 3.9 testing.

You're right that allowing the test CI to use the latest versions of dask, numpy and scipy might give us an earlier indication of compatibility problems, but to be honest we just don't have the bandwidth over at this project to be proactive about that. In addition, the CI is typically run quite infrequently here (like months apart) and there can be a *lot* of changes with other library updates in the interim, so when any failures due to package updates do happen it will be very difficult to narrow down where the problem is coming from. Those constraints are the reason for the decision here.

Reply:
Agree with Genevieve. The initial idea of pinnings here was to minimize surprising CI failures due to dependency updates.

Understand why pinnings for Python 3.8 wouldn't copy over to Python 3.9. That said, it would be good to find a set of frozen dependencies that do make sense for Python 3.9 and include them here.

Also would be good to keep a set of frozen dependencies for other Python versions tested. Optionally we may take a moment here to refresh them.

Reply:
Thanks for the detailed explanation @GenevieveBuckley - in that case, I will go ahead and pin the 3.9 environment to the versions picked up in the environment creation.

> Optionally we may take a moment here to refresh them.

Happy to do that here if we're okay with keeping that all in one PR 🙂 also while we're refreshing the pinnings should I go ahead and pin `slicerator` since that was unpinned up until now?

Reply:
> should I go ahead and pin `slicerator` since that was unpinned up until now?

I don't recall the full history here, but we did use to have this pinned and that changed in PR ( https://github.com/dask/dask-image/pull/188 ). Perhaps there was some issue with pinning previously?

Would think we would want that pinned, but will defer to @GenevieveBuckley on this as she may know more 🙂

Reply:
The reason slicerator is unpinned appears to be that when I was trying to make the github actions CI run, the pinning was causing me problems. There is a commit here from me called ["Yep, we really do need slicerator unpinned "](https://github.com/dask/dask-image/pull/188/commits/47b242ae607ec0a905451d2e4d180e7832e3b419), so you can see I tried slicerator both pinned and unpinned in PR https://github.com/dask/dask-image/pull/188

Mind you, I just unpinned it entirely, instead of trying to work out what other pinned version would work. I cut corners to finish the job quicker.

Reply:
Thanks for the context! Things seem to be working now with it pinned to 1.0.0, so we should be safe to proceed with it pinned from now on